content/troubleshooting/common-issues.md

```diff
@@ -4,19 +4,18 @@ title: Common issues
 weight: 1
 ---
 
-## Two-factor authentication for Apple Developer Portal integration fails
+## iOS errors regarding creating authenticated sessions
 
-Recently, Apple changed their private API which affects the way third-party systems offer two-factor authentication for Apple Developer Portal. This may result in two-factor authentication failing for the Apple Developer Portal integration in Codemagic despite entering a correct authentication code.
+When App Store Connect is not correctly set up, users might encounter an error message similar to the following:
 
-        Apple Developer Portal authentication failed
-        Two factor authentication failed for user@domain.com: Incorrect Verification Code: Incorrect verification code
-
-If you see the error message above and are convinced you have entered the correct authentication code, you can try the following workarounds:
+```
+altool[xxx:xxx] *** Error: Unable to validate archive '/Users/builder/ipas/xxx'.
+altool[xxx:xxx] *** Error: code -22020 (Unable to validate your application. We are unable to create an authentication session.)
+```
 
- * Change your Apple Developer Portal password
- * Log out from all the devices except the one you want to use for receiving the authentication code
+The most common reason for the occurrence of this error message is that the user has not properly set up his/hers App-specific password and has used the Apple Developer account password instead.
```

Comment:
The most common reason for the occurrence of this error message is using Apple ID password instead of **app-specific password** in App Store Connect publishing settings.

Reply:
app-specific password can be linked there: https://support.apple.com/en-us/HT204397